mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMailboxMapper.java

```diff
@@ -134,8 +135,8 @@ public void endRequest() {
     /**
      * @see org.apache.james.mailbox.store.mail.MailboxMapper#hasChildren(org.apache.james.mailbox.store.mail.model.Mailbox, char)
      */
-    public boolean hasChildren(Mailbox mailbox, char delimiter) throws MailboxException {
-        String mailboxName = mailbox.getName() + delimiter;
+    public boolean hasChildren(Mailbox mailbox, PathDelimiter delimiter) throws MailboxException {
+        String mailboxName = delimiter.appendDelimiter(mailbox.getName());
```

Comment:
could be replaced by a matching method instead of startsWith

Reply:
Yes I'm thinking to MailboxUtil::isChildOf....